plugins/woocommerce/includes/admin/class-wc-admin-brands.php

```diff
@@ -370,6 +370,23 @@ public function add_thumbnail_field() {
 						jQuery('.remove_image_button').hide();
 						return false;
 					});
+
+					jQuery( document ).on( 'ajaxComplete', function( event, request, options ) {
+						if ( request && 4 === request.readyState && 200 === request.status
+							&& options.data && 0 <= options.data.indexOf( 'action=add-tag' ) ) {
+
+							var res = wpAjax.parseAjaxResponse( request.responseXML, 'ajax-response' );
+							if ( ! res || res.errors ) {
+								return;
+							}
+							// Clear Thumbnail fields on submit
+							jQuery( '#product_cat_thumbnail' ).find( 'img' ).attr( 'src', '<?php echo esc_js( wc_placeholder_img_src() ); ?>' );
+							jQuery( '#product_cat_thumbnail_id' ).val( '' );
+							jQuery( '.remove_image_button' ).hide();
```

Comment:
> Unify the additional taxonomy management code with product categories

Yeah this is an opportunity and given they're in separate scripts/scopes I wouldn't invest in unifying it.

However, this part is nearly exactly the same as `jQuery(document).on( 'click', '.remove_image_button',` callback within the same scope. How about extracting it to the function (something like `clearThumbnailField`) and reuse it in both places?